ramanujan/enumerators/AbstractGCFEnumerator.py

```diff
@@ -0,0 +1,190 @@
+        self.verify_dps = g_N_verify_dps  # working decimal precision for validating results.
+        self.const_sym = sym_constants
+        self.constants_generator = create_mpf_const_generator(sym_constants)
+        
+        # expand poly domains object
+        # there are two methods to generate and iter over domains.  the newer one uses poly_domains_generator only,
+        # but the old one still uses the rest of the arguments.
+        # generating them here to avoid breaking older enumerators
+        self.poly_domains_generator = poly_domains_generator
+        a_iterator_func, b_iterator_func = poly_domains_generator.get_calculation_method()
+        self.create_an_series = \
+            lambda coefs, items: get_series_items_from_iter(a_iterator_func, coefs, items)
+        self.create_bn_series = \
+            lambda coefs, items: get_series_items_from_iter(b_iterator_func, coefs, items)
+        self.get_an_length = poly_domains_generator.get_an_length
+        self.get_bn_length = poly_domains_generator.get_bn_length
+        
+        self.get_an_iterator = poly_domains_generator.get_a_coef_iterator
+        self.get_bn_iterator = poly_domains_generator.get_b_coef_iterator
+        
+        # store lhs_hash_table
+        self.hash_table = hash_table
+
+    def __get_formatted_results(self, results: List[RefinedMatch]) -> List[FormattedResult]:
+        ret = []
+        for r in results:
+            an = self.create_an_series(r.rhs_an_poly, 250)
+            bn = self.create_bn_series(r.rhs_bn_poly, 250)
+            print_length = max(max(get_size_of_nested_list(r.rhs_an_poly), get_size_of_nested_list(r.rhs_bn_poly)), 5)
+            gcf = GeneralizedContinuedFraction(an, bn[1:])
+            sym_lhs = self.hash_table.evaluate_sym(r.lhs_key, self.const_sym)[r.lhs_match_idx]
+            ret.append(FormattedResult(sym_lhs, gcf.sym_expression(print_length), gcf))
+        return ret
+
+    def __get_formatted_polynomials(self, result: RefinedMatch):
+        def sym_poly(poly_deg, poly_terms):
+            poly = list(reversed(find_polynomial_series_coefficients(poly_deg, poly_terms, 0)))
+            n = sympy.Symbol('n')
+            poly_sym = 0
+            for i in range(len(poly)):
+                poly_sym += n ** i * poly[i]
+            return poly_sym
+
+        an_poly_max_deg = get_size_of_nested_list(result.rhs_an_poly)
+        an = self.create_an_series(result.rhs_an_poly, an_poly_max_deg + 1)
+        bn_poly_max_deg = get_size_of_nested_list(result.rhs_bn_poly)
+        bn = self.create_bn_series(result.rhs_bn_poly, bn_poly_max_deg + 1)
+        an_eq = sympy.Eq(sympy.Symbol('a(n)'), sym_poly(an_poly_max_deg, an))
+        bn_eq = sympy.Eq(sympy.Symbol('b(n)'), sym_poly(bn_poly_max_deg, bn))
+        return an_eq, bn_eq
+
+    def print_results(self, results: List[RefinedMatch], latex=False, convergence_rate=True):
```

Comment:
in the tests, we might want to use convergence_rate=False as it slows down the tests significantly.

Reply:
changed